packages/waku/src/lib/plugins/vite-plugin-fs-router-typegen.ts

```diff
@@ -6,49 +6,29 @@ import { joinPath } from '../utils/path.js';
 
 const SRC_PAGES = 'pages';
 
+const invalidCharRegex = /[^0-9a-zA-Z_$]/g;
```

Comment:
To be correct... it looks like unicode chars might be allowed. And there are reserved words that need to be replaced too if they match exactly. https://tc39.es/ecma262/multipage/ecmascript-language-lexical-grammar.html#sec-names-and-keywords

Reply:
sure, thanks! I meant a comment in the code with the regex though 😅 - just for anyone looking later without the PR context handy

Reply:
Right. It's a good suggestion. I agree we should document it. I just wanted to chat about that first before making revisions.

I found this lib - https://github.com/eemeli/safe-identifier. It references a blog article that doesn't exist anymore that suggests appending hash of the original var name to the sanitized name to ensure that it will be unique. https://web.archive.org/web/20110703045355/https://werxltd.com/wp/2010/05/13/javascript-implementation-of-javas-string-hashcode-method/

I think I'll update it to be more correct by using a technique like that.

Reply:
I also found this reference.
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#identifiers
https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Lexical_grammar#reserved_words

We reserved words are not an issue because we uppercase the first char.